deployments/gpu-operator/reconfigure-mig.go

```diff
@@ -0,0 +1,982 @@
+	if [ "${?}" != "0" ]; then
+		echo "Unable to set the value of 'nvidia.com/mig.config.state' to 'rebooting'"
+		echo "Exiting so as not to reboot multiple times unexpectedly"
+		exit_failed
+	fi
+	chroot ${HOST_ROOT_MOUNT} reboot
+	exit 0
+
+fi
+*/
+func applyMIGModeChange(nvidiaMigPartedAlias string, migConfigFile string, selectedMIGConfig string,
+	 nodeName string, withReboot bool, hostRootMount string, currentLabels map[string]string, hostGPUClientServices []string, withShutdownHostGPUClients bool) error {
+	applyCmd := exec.Command(nvidiaMigPartedAlias, "-d", "apply", "--mode-only", "-f", migConfigFile, "-c", selectedMIGConfig)
+	assertCmd := exec.Command(nvidiaMigPartedAlias, "-d", "assert", "--mode-only", "-f", migConfigFile, "-c", selectedMIGConfig)
+
+	if err := applyCmd.Run(); err != nil {
+		fmt.Println("Failed to apply MIG mode change: ", err)
+		if withReboot {
+			rebootNode(nodeName, hostRootMount, currentLabels, hostGPUClientServices, withShutdownHostGPUClients)
+		}
+		return err
+	}
+
+	if err := assertCmd.Run(); err != nil {
+		fmt.Println("Failed to assert MIG mode change: ", err)
+		if withReboot {
+			rebootNode(nodeName, hostRootMount, currentLabels, hostGPUClientServices, withShutdownHostGPUClients)
+		}
+		return err
+	}
+	fmt.Println("MIG mode change applied successfully")
+	return nil
+}
+
+func rebootNode(nodeName string, hostRootMount string, currentLabels map[string]string, hostGPUClientServices []string, withShutdownHostGPUClients bool) {
+
+	fmt.Println("Changing the 'nvidia.com/mig.config.state' node label to 'rebooting'")
+	cmd := exec.Command("kubectl", "label", "--overwrite", "node", nodeName, "nvidia.com/mig.config.state=rebooting")
+	if err := cmd.Run(); err != nil {
+		fmt.Println("Unable to set the value of 'nvidia.com/mig.config.state' to 'rebooting'")
+		fmt.Println("Exiting so as not to reboot multiple times unexpectedly")
+		setStateAndExit("failed", 1, currentLabels, hostGPUClientServices, withShutdownHostGPUClients, hostRootMount, nodeName)
+	}
+
+	cmdRoot := exec.Command("chroot", hostRootMount, "reboot")
+	if err := cmdRoot.Run(); err != nil {
+		fmt.Println("Failed to reboot the node: ", err)
+		os.Exit(0)
+	}
+}
+
+/*
+	echo "Applying the selected MIG config to the node"
+	nvidia-mig-parted -d apply -f ${MIG_CONFIG_FILE} -c ${SELECTED_MIG_CONFIG}
+	if [ "${?}" != "0" ]; then
+		exit_failed
+	fi
+*/
+
+func applyMigConfig(nvidiaMigPartedAlias string, migConfigFile string, selectedMIGConfig string, currentLabels map[string]string, hostGPUClientServices []string, withShutdownHostGPUClients bool, hostRootMount string, nodeName string) error {
+	fmt.Println("Applying the selected MIG config to the node")
+	cmd := exec.Command(nvidiaMigPartedAlias, "-d", "apply", "-f", migConfigFile, "-c", selectedMIGConfig)
+	if err := cmd.Run(); err != nil {
+		setStateAndExit("failed", 1, currentLabels, hostGPUClientServices, withShutdownHostGPUClients, hostRootMount, nodeName)
```

Comment:
moved the setState() function to the reconfigure() function. But, that didn't help in reducing the number of parameters. Since setState() function need all these parameters in isolation itself.